server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/Main.scala

```diff
@@ -17,27 +18,28 @@ import org.broadinstitute.dsp.workbench.welder.JsonCodec._
 import org.http4s.server.blaze.BlazeServerBuilder
 
 import scala.concurrent.ExecutionContext
-import scala.concurrent.duration._
 
 object Main extends IOApp {
   override def run(args: List[String]): IO[ExitCode] = {
     implicit val unsafeLogger = Slf4jLogger.getLogger[IO]
 
     val app: Stream[IO, Unit] = for {
+      blockingEc <- Stream.resource[IO, ExecutionContext](ExecutionContexts.fixedThreadPool(255))
+      implicit0(it: Linebacker[IO])  = Linebacker.fromExecutionContext[IO](blockingEc)
       appConfig <- Stream.fromEither[IO](Config.appConfig)
-      blockingEc <- Stream.resource[IO, ExecutionContext](ExecutionContexts.fixedThreadPool(255)) //TODO: use cached thread pool with some back pressure
-      storageLinksCache <- cachedResource[LocalDirectory, StorageLink](appConfig.pathToStorageLinksJson, blockingEc, storageLink => List(storageLink.localBaseDirectory -> storageLink, storageLink.localSafeModeBaseDirectory -> storageLink))
+      storageLinksCache <- cachedResource[Path, StorageLink](appConfig.pathToStorageLinksJson, blockingEc, storageLink => List(storageLink.localBaseDirectory.path -> storageLink, storageLink.localSafeModeBaseDirectory.path -> storageLink))
       metadataCache <- cachedResource[Path, GcsMetadata](appConfig.pathToGcsMetadataJson, blockingEc, metadata => List(metadata.localPath -> metadata))
-      _ <- Stream.eval(timer.sleep(2 seconds)) // sleep 2 seconds to wait for google application default credential becomes available
```

Comment:
can you fill me in on what the solution for this was?

Reply:
it was when Joel was working on deploying welder via leo and it can't find application default, yet it can find it if we restart welder service, so at the time, we thought a short delay before reading the application default will be good enough to make application default available on the box, but Joel ended doing what jupyter container is doing, which is overwriting startup script to a dummy command, and start welder service at some point later, which worked.